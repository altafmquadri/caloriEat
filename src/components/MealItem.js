import React from 'react';
import { Link } from 'react-router-dom'

export const MealItem = (props) => {
    return (
        <div>
            <Link to={{
                pathname: `/edit/${props.uuid}`,
                date: props.date
            }}>
                <h2>{props.foodItem}</h2>
            </Link>
            <h3>{props.calories}</h3>
        </div>
    )
}

export default MealItem