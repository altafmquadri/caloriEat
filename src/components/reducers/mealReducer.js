const mealReducer = (state = [], action) => {
    switch (action.type) {
        case 'ADD_MEAL': {
            const meal = action.payload
            return [...state, meal]
        }
        case 'EDIT_MEAL': {
            const {
                calories,
                totalNutrients: {
                    CHOCDF: { quantity: carbohydrates },
                    FAT: { quantity: fat },
                    FIBTG: { quantity: fiber },
                    PROCNT: { quantity: protein },
                }
            } = action.payload
            const item = {
                ...state.find(meal => meal.uuid === action.uuid),
                ...action.properties,
                calories,
                carbohydrates: parseFloat(carbohydrates).toFixed(2),
                fat: parseFloat(fat).toFixed(2),
                fiber: parseFloat(fiber).toFixed(2),
                protein: parseFloat(protein).toFixed(2),
            }
            return state.map(meal => {
                if (meal.uuid === action.uuid) {
                    return meal = { ...meal, ...item }
                }
                return meal
            })
        }
        case 'DELETE_MEAL': {
            const id = action.payload
            return state.filter(meal => meal.uuid !== id)
        }
        default:
            return state
    }
}

export default mealReducer