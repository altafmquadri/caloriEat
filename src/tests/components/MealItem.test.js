import React from 'react';
import { shallow } from 'enzyme'
import MealItem from '../../components/MealItem'
import { meals } from '../fixtures/mealReducer'

describe('MealItem />', () => {
    test('should render MealItem', () => {
        const wrapper = shallow(<MealItem {...meals[0]} />)
        expect(wrapper).toMatchSnapshot()
    })
})