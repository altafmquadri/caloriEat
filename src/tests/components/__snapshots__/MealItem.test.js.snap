// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealItem /> should render MealItem 1`] = `
<div>
  <Link
    to={
      Object {
        "date": "2020-08-17T18:36:13.173Z",
        "pathname": "/edit/1",
      }
    }
  >
    <h2>
      milk
    </h2>
  </Link>
  <h3>
    148
  </h3>
</div>
`;
